src/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs

```diff
@@ -306,6 +306,29 @@ public static async Task Zip64ArchiveEntry_CorruptedFile_Read_UpToUncompressedSi
             }
         }
 
+
+        [Fact]
+        public static async Task UnseekableVeryLargeArchive_DataDescriptor_Read_Zip64()
+        {
+            MemoryStream stream = await LocalMemoryStream.readAppFileAsync(strange("veryLarge.zip"));
+
+            using (ZipArchive archive = new ZipArchive(stream, ZipArchiveMode.Read))
+            {
+                ZipArchiveEntry e = archive.GetEntry("bigFile.bin");
+                using (Stream source = e.Open())
+                {
+                    byte[] buffer = new byte[s_bufferSize];
+                    int read;
+                    while ((read = source.Read(buffer, 0, buffer.Length)) != 0)
```

Comment:
I believe this code still has an issue.  Typically the contract for Read doesn't require a stream to fill the entire buffer so its possible the stream implementation might always return less than the buffer size for the entire length of the file (though granted its unlikely). 

I wonder what case you're ever expecting to loop here.  Can you either eliminate the looping and just assert that you read some bytes, or loop until you fill a single buffer and assert that you were able to do that?

Reply:
> Typically the contract for Read doesn't require a stream to fill the entire buffer so its possible the stream implementation might always return less than the buffer size for the entire length of the file (though granted its unlikely)

Yes it is granted that DeflateStream.Read(...) will fill entire buffer as it would fill less only in case inflated stream has less bytes and for this test uncompressed file has much larger size than this buffer 

> I wonder what case you're ever expecting to loop here. Can you either eliminate the looping and just assert that you read some bytes, or loop until you fill a single buffer and assert that you were able to do that?

My initial intent was to loop only twice to show it is reads successfully. But yes you are right, no need to loop here, one read is actually enough